vars/apmCli.groovy

```diff
@@ -70,36 +56,23 @@ def call(Map args = [:]) {
       }
     }
 
-    withEnvMask(vars: [
-      [var: "APM_CLI_SERVER_URL", password: "${url}"],
-      [var: "APM_CLI_TOKEN", password: "${token}"],
-      [var: "APM_CLI_SERVICE_NAME", password: "${serviceName}"],
-      [var: "APM_CLI_TRANSACTION_NAME", password: "${transactionName}"],
-      [var: "APM_CLI_SPAN_NAME", password: "${spanName}"],
-      [var: "APM_CLI_SPAN_COMMAND", password: "${spanCommand}"],
-      [var: "APM_CLI_SPAN_LABELS", password: "${spanLabel}"],
-      [var: "APM_CLI_PARENT_TRANSACTION_SAVE", password: "${ saveTsID ? 'tsID.txt' : ''}"],
-      [var: "APM_CLI_TRANSACTION_RESULT", password: "${result}"],
-      [var: "APM_CLI_PARENT_TRANSACTION", password: "${parentTransaction}"],
-    ]){
-      log(level: 'DEBUG', text: "apmCli: Runninf APM CLI")
-      sh(script: """#!/bin/bash +x
-            if [ -z "\$(command -v python3)" ] \
-              || [ -z "\$(command -v virtualenv)" ]; then
-              exit 0
-            fi
-            virtualenv -q --python=python3 .venv
-            source ".venv/bin/activate"
-            pip -q install -r requirements.txt
-            set -x
-            export |grep APM_CLI
-            python3 apm-cli.py
-          """,
-        label: 'apm-cli'
-      )
-      if(saveTsID){
-        log(level: 'DEBUG', text: "apmCli: Persistent transaction ID on APM_CLI_PARENT_TRANSACTION")
-        setEnvVar('APM_CLI_PARENT_TRANSACTION', readFile(file: 'tsID.txt'))
+    withAPMService(args){
+      withEnv([
+        "APM_CLI_SERVICE_NAME=${serviceName}",
+        "APM_CLI_TRANSACTION_NAME=${transactionName}",
+        "APM_CLI_SPAN_NAME=${spanName}",
+        "APM_CLI_SPAN_COMMAND=${spanCommand}",
+        "APM_CLI_SPAN_LABELS=${spanLabel}",
+        "APM_CLI_PARENT_TRANSACTION_SAVE=${ saveTsID ? 'tsID.txt' : ''}",
+        "APM_CLI_TRANSACTION_RESULT=${result}",
+        "APM_CLI_PARENT_TRANSACTION=${parentTransaction}",
+      ]){
+        log(level: 'DEBUG', text: "apmCli: Runninf APM CLI")
+        python(label: 'apmCLI', file: 'apm-cli.py')
```

Comment:
less code :D

Reply:
should this particular step trap the errors to avoid failing the build if something bad happened?